platforms/netbsd/hax_wrapper.c

```diff
@@ -251,36 +251,34 @@ hax_atomic_t hax_atomic_dec(volatile hax_atomic_t *atom)
 
 int hax_test_and_set_bit(int bit, uint64_t *memory)
 {
-    volatile uint64_t *val;
-    uint64_t mask, old;
-
-    val = (volatile uint64_t *)memory;
-    mask = 1 << bit;
+    const unsigned int units = (sizeof(*memory) * CHAR_BIT);
+    volatile unsigned long *const p = &memory[bit / units];
```

Comment:
nit: Just curious, what's the benefit of using `unsigned long` and `atomic_cas_ulong` instead of `uint64_t` and `atomic_cas_64`? I don't think the following code would work correctly if `sizeof(unsigned long) != sizeof(uint64_t)`.

Reply:
`ulong` is more portable and since we need to support >63 bit it's better to stick to it. In future it can work better on other CPUs that don't implement 64-bit atomics or just don't want to use compiler flags to get them to use like on 32-bit CPUs where we need i586/i686 target)

Reply:
That's a good point. But this line does an implicit cast from `uint64_t *` to `ulong *`. Shouldn't we make `units` match `sizeof(ulong)`? Something like:

```c
    unsigned long *p0 = (unsigned long *)memory;
    const unsigned int units = sizeof(*p0) * CHAR_BIT;
    volatile unsigned long *const p = &p0[bit / units];
```

Reply:
I've decided to keep for now atomic_cas_64 and don't do trick on bits unless we can test it on a big endian setup.